### PR TITLE
Data storage

### PR DESCRIPTION
**Pull request by Thoth1111**
### Data Storage

In this branch, I created an object to store information from the contact form and wrote a function to store and retrieve it. The steps included:

- Creating an object
-Pushing each input value into the object
-Stringifying the values and adding them to local storage.
Parsing & retrieving the values.

### CODE REVIEW

**Comment by Thoth1111** [comment]
Please note that this is my first review request. There appears to be an error that's making my coding partner's review request count as my own. It now says this is my 3rd review request when in actual sense, it is my first.

File: myscript.js [hunk 1]
```diff
@@ -141,3 +141,26 @@ form.addEventListener('submit', (e) => {
...
+    dataInput[input.message] = input.value;
+    localStorage.setItem('grouped', JSON.stringify(dataInput));
+  });
+});
+const archived = JSON.parse(localStorage.getItem('grouped'));
+if (archived) {
+  inputs.forEach((element) => {
+    element.value = archived[element.name];
+  });
+}
```
Comment (Tufahel) [inline myscript.js:166]: - [ ] Data is not saving in local storage correctly. After `reloading` the page, the data inside the form is missing. Kindly fix this issue.
  - **Your website:**
   ezgif com-gif-maker (5)
  - **Data in Local storage should look like this:**
  Screenshot from 2022-12-24 04-38-30
  - You can consider using `window.onload()` function to get the data from local storage after loading and correctly saving data in local storage. An example is given below:
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
}
```

**Review by Tufahel — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Thoth1111, @HanaSabih , :wave:  

Alt Text

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :runner: 

## Highlights 🌟

✔️ Linters checks are passing
✔️ Use GitHub Flow
✔️ Meaningful commit messages.

## Required Changes ♻️

_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tufahel** [comment]
> Please note that this is my first review request. There appears to be an error that's making my coding partner's review request count as my own. It now says this is my 3rd review request when in actual sense, it is my first.

- Hi @Thoth1111 :wave:,  As mentioned in the [requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md), this is a group project. So, if anyone from your group asks for a review, it will count for all members; it's not an error.
- You can see this requirement on **Work and submission mode**.
Screenshot from 2022-12-24 04-46-02

**Pull request closed without merging**